lighthouse-core/lib/network-request.js

```diff
@@ -341,10 +363,89 @@ module.exports = class NetworkRequest {
     // Bail if we somehow already have transfer size data.
     if (!global.isLightrider) return;
 
-    const totalFetchedSize = this.responseHeaders.find(item => item.name === 'X-TotalFetchedSize');
+    const totalFetchedSize = this.responseHeaders.find(item => item.name === HEADER_FETCHED_SIZE);
     // Bail if the header was missing.
     if (!totalFetchedSize) return;
-    this.transferSize = parseFloat(totalFetchedSize.value);
+    const floatValue = parseFloat(totalFetchedSize.value);
+    // Bail if the header cannot be parsed
+    if (isNaN(floatValue)) return;
+    this.transferSize = floatValue;
+  }
+
+  /**
+   * TODO(exterkamp) add explanatory comment.
+   */
+  _updateTimingsForLightrider() {
+    // Bail if we aren't in Lightrider.
+    if (!global.isLightrider) return;
+
+    // Bail if timing is not initialized.
+    if (!this.timing) {
+      return;
+    }
+
+    // For more info on timing nomenclature: https://www.w3.org/TR/resource-timing-2/#processing-model
+
+    //    StartTime
+    //    | ConnectStart
+    //    | |     SSLStart  SSLEnd
+    //    | |     |         | ConnectEnd
+    //    | |     |         | | SendStart/End   ReceiveHeadersEnd
+    //    | |     |         | | |               |                EndTime
+    //    ▼ ▼     ▼         ▼ ▼ ▼               ▼                ▼
+    //    [ [TCP  [   SSL   ] ] [   Request   ] [   Response   ] ]
+    //    ▲ ▲     ▲         ▲ ▲ ▲             ▲ ▲              ▲ ▲
+    //    | |     '-SSLMs---' | '-requestMs---' '-responseMs---' |
+    //    | '----TCPMs--------'                                  |
+    //    |                                                      |
+    //    '------------------------TotalMs-----------------------'
+
+    const totalHeader = this.responseHeaders.find(item => item.name === HEADER_TOTAL);
+    // Bail if there was no totalTime.
+    if (!totalHeader) return;
+
+    const totalMs = parseInt(totalHeader.value);
+
+    const TCPHeader = this.responseHeaders.find(item => item.name === HEADER_TCP);
+    const requestHeader = this.responseHeaders.find(item => item.name === HEADER_REQ);
+    const SSLHeader = this.responseHeaders.find(item => item.name === HEADER_SSL);
+    const responseHeader = this.responseHeaders.find(item => item.name === HEADER_RES);
+
+    // Make sure all Times are initialized and are non-negative.
+    const TCPMs = Math.max(0, TCPHeader === undefined ? 0 : parseInt(TCPHeader.value));
```

Comment:
what about
`const TCPMs = TCPHeader === undefined ? 0 : Math.max(0, parseInt(TCPHeader.value));`

but can also use falsiness of `undefined` with
`const TCPMs = TCPHeader ? Math.max(0, parseInt(TCPHeader.value)) : 0;`

Reply:
should we explicitly worry about `NaN` here (`isNaN(Math.max(0, NaN)) === true`)? Or is the implicit check in the timings adding up sufficient (good thing `NaN !== NaN`)? I worry that we'll forget at some point

Reply:
Changed to second rec.  Added a comment about NaN, but I don't think its a huge problem.